enumerables.rb

```diff
@@ -0,0 +1,155 @@
+module Enumerable
+    def my_each
+        for i in 0...self.length do
+            yield(self[i])
+        end
+    end
+    
+    def my_each_with_index
+        for i in 0...self.length do
+            yield(self[i], i)
+        end
+    end
+
+    def my_select
+        a = []
+        my_each do |n|
+            if yield(n)
+                a.push(n)
+            end
+        end
+        a
+    end
+
+    def my_all?
+        my_each do |n|
+            if !yield(n)
+                return false
+            end
+        end
+        true
+    end
+
```

Comment:
- [ ] should have working `my_all?` identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))

- [ ] my_all should return false if the block returns false or nil::range, also you are not handling classes, regex and patterns correctly.